generator/spring-boot/src/main/java/io/fabric8/maven/generator/springboot/SpringBootGenerator.java

```diff
@@ -124,12 +124,36 @@ protected boolean isFatJar() throws MojoExecutionException {
         List<String> answer = new ArrayList<>();
         Properties properties = SpringBootUtil.getSpringBootApplicationProperties(this.getProject());
         String port = properties.getProperty(SpringBootProperties.SERVER_PORT, DEFAULT_SERVER_PORT);
-        addPortIfValid(answer, getConfig(JavaExecGenerator.Config.webPort, port));
+        addPortIfValid(answer, getConfig(JavaExecGenerator.Config.webPort, replaceKeyWithSystemPropertyValue(port)));
         addPortIfValid(answer, getConfig(JavaExecGenerator.Config.jolokiaPort));
         addPortIfValid(answer, getConfig(JavaExecGenerator.Config.prometheusPort));
         return answer;
     }
 
+    /**
+     *  Look within all the System Properties if there is a property which corresponds to the key passed
+     *  If a match exists, then replace the value otherwise use the default value defined within ${key:default-value}
+     *  Example
+     *  my.server.port=7777 is a system property that we will search in order to replace its value within the key passed to the method.
+     *  If the key is "server.port: ${my.server.port:6666}", then "my.server.port" value will be substituted, otherwise the default value is used
+     *  "6666"
+     *
+     *  @param key The string containing the key to search
+     *  @return String The modified key
+     */
+    protected static String replaceKeyWithSystemPropertyValue(String key) {
```

Comment:
As Nicola proposed also that we support maven properties and system properties where maven props override system props, I will update the PR accordingly.

Nicola suggested too to move the code of the method `replaceKeyWithSystemPropertyValue` to the class SpringBootUtil packaged within f-m-p core. Do you agree too @nicolaferraro @rhuss ?